extends Control

@onready var jugar_button = $VBoxContainer/Jugar
@onready var cargar_button = $VBoxContainer/Cargar
@onready var opciones_button = $VBoxContainer/Opciones
@onready var salir_button = $VBoxContainer/Salir
@onready var music_player = $MusicPlayer  # ðŸ‘ˆ AsegÃºrate que el nodo se llame asÃ­

@onready var options_scene := preload("res://scenes/options_menu.tscn") 
var options_instance: Node = null

func _ready():
	if music_player and not music_player.playing:
		music_player.play()
		music_player.stream_paused = false
		print("ðŸŽ¶ MÃºsica iniciada")
	# Desactiva el botÃ³n de cargar por defecto (hasta implementar el guardado)

	# Conectar seÃ±ales de los botones
	jugar_button.pressed.connect(_on_jugar_pressed)
	cargar_button.pressed.connect(_on_cargar_pressed)
	opciones_button.pressed.connect(_on_opciones_pressed)
	salir_button.pressed.connect(_on_salir_pressed)

func _on_jugar_pressed():
	get_tree().change_scene_to_file("res://scenes/Levels/stage_1.tscn")


func _on_cargar_pressed():
	var load_scene = preload("res://scenes/load_menu.tscn").instantiate()
	add_child(load_scene)



func _on_opciones_pressed():
	if not options_instance:
		options_instance = options_scene.instantiate()
		add_child(options_instance)
	else:
		options_instance.visible = true

func _on_salir_pressed():
	get_tree().quit()
